docs/DeveloperGuide.md

```diff
@@ -1,38 +1,150 @@
+
+The ***Architecture Diagram*** given above presents the high-level design of the product. The main components included are as follows:
+1. `Duke`: Initialised when launch and acts as the main logic component of the program.
+2. `UI`: Handles all user interaction, including printing responses and receiving user inputs.
+3. `Parser`: Parses main user commands and calls for appropriate functions.
+4. `commands`: Exists as a package which includes all command functions.
+5. `Timetable`: Contains all the data in the current session. (Data can only be accessed and manipulated by commands)
+6. `data`: Manages the saving and loading of the current session data.
+7. `Nusmods`: Handles all communication to the NUSMods API and extracts useful information that is used for the rest of
+   the program.
+
+<p style='text-align: justify;'>The design of the product focuses on low coupling and high cohesion. Following as closely to the Single Responsibility Principle (SRP) as possible, each major component explained above has unique roles which is not overlapped by another component. For example, printing of all responses go through the UI class, which handles all interactions with user, including the display of program responses.</p>
+
+---
+
+#### Key Features
+##### Adding modules to current timetable
+<img src="images/addModule.png" width="580" />
+
+The ***Sequence Diagram*** above is a simplified depiction of how new modules are added to the user current session timetable. The feature is rather complex due to the need to query, extract, and check the data before adding it to the timetable. Thus, a few design decisions were made:</p>
+
+* The class CommandAddModule manages the key actions (as explained above) from a higher level.
+* The class Nusmods manages all interactions with the NUSmods API and deals with JsonNode objects.
+* Any particular module is only added to the current session timetable through the Timetable class after all relevant data has been extracted, packaged and checked.
+* Module object consists of all relevant data for the module, including code, name, details, all available lessons, and lessons that is to be attented by the user.
+
+<p style='text-align: justify;'>The class CommandAddModule provides control abstraction to the comlpex operation as it abstracts away lower level data items such as the day and time of a specific tutorial class for a particular module.</p>
+
+<p style='text-align: justify;'>By having a separate class Nusmods deal with all API calls and JsonNode objects, it allows the other classes/components to only have to work with simpler data types and native objects (such as Module and Lesson objects).</p>
+
+<p style='text-align: justify;'>Other alternatives considered includes having the CommandAddModule class handle the API call but it would have resulted in the code being even more convoluted and less readable.</p>
```

Comment:
Do consider adding a design decision label similar to data section below. This is because this portion is no longer explaining about the key features but rather the rationale in choosing the key features.